### DISCUSSION TITLE
What label printer do people use?

### DISCUSSION THREAD

**Discussion by afsabresfan** [Donkie/Spoolman#425] [General]
I'm just getting started with Spoolman but I've already got some partial rolls of filament I want to sort through and label. I'm just curious what kind of label printer people are using. I've seen some generally referenced like "thermal printer", but I was curious if anyone is running a printer connected to a Windows machine that prints right from the Spoolman printer interface. Thanks!

**Comment by Donkie** [comment]
Personally I just use some A4 label paper on a regular laser printer. Works well enough for me

**Comment by afsabresfan** [comment]
I bought a Nelko PM220 on sale on Amazon. Took a little bit of work to get it set up correctly, but I got it to print really nice size labels! The only thing I can’t tell yet is whether or not the QR code is big enough, as I haven’t set up my reverse proxy yet. I’ll report back when that’s done.

The key to the PM220 is to force the computer to print directly to the printer, not using the spooler. It’s in the printer settings. Once you do that, it prints great!

**Reply by sdwilsh** [reply]
@afsabresfan, could you share more details about your setup for this?

**Comment by mfraser** [comment]
I have a Dymo Labelwriter 450 and am hoping to use that to print the labels now that I have set up OctoEverywhere.

**Comment by pdsccode** [comment]
Running a Brother P-Touch P750W here for quite some time. The 24mm labels are my go-to. Easy to setup, portable and compatible with every device I use.

**Comment by TheAner** [comment]
I bought recently Niimbot d11 and it comes with 60x12mm foiled stickers (you could buy different).
I created a label preset in spoolman for it and everything works well.

**Reply by Ken6432** [reply]
How do you get the label information from the web interface of Spoolman to the Niimbot.  I have a d101 and can't see how to do that.  It seems like everything I print has to done in the Niimbot phone app.  Am I missing something obvious? -- Thanks

**Reply by TheAner** [reply]
I don't, I just used label size information from the nimbot app to create label preset (multiplied by 2 to have the same ratio, but double the resolution) and export the image. Then I print the labels by adding image file in niimbot app and print it there.

![obraz](https://github.com/user-attachments/assets/14fd3d4c-a836-4e23-aa6c-eb521d431ad7)
![obraz](https://github.com/user-attachments/assets/2d72c9fa-79f0-4b33-baa3-f26b5852ef7b)

**Reply by Ken6432** [reply]
OK, thanks!  I was hoping I missed some easy way to get them to print.  I like the Niimbot, but I wish it was a bit more flexible to print things to

**Reply by Ken6432** [reply]
@TheAner - I managed to create a custom paper size in my print to PDF driver in Windows.  Now I can print the labels to a PDF with a 60x12 page size.  I can then import that PDF to the Niimbot printer software and print the labels.  It works well, but it sure was a pain to configure

**Reply by TheAner** [reply]
@Ken6432 

But why are you downloading it as a pdf instead of saving it as a image?

**Reply by Ken6432** [reply]
Images were not as clear from as the PDFs are

**Reply by TheAner** [reply]
That's why I bumped up resolution with the same proportions and now it's sharp

**Comment by Play13thStreet** [comment]
I use Label Paper A4 HERMA 10906, works perfect.

**Comment by mtabo** [comment]
I have a HPRT HT300, I print on 50x25 labels the images that Spoolman gives but they are not sharp, I would like to have support for ZPL but I don't have the time at the moment to implement it.

**Comment by Rin010** [comment]
Export to Excel using Python
Create labels using ZebraDesigner
![image](https://github.com/user-attachments/assets/8aa369d8-7788-4a8b-b8cf-e7702d5df5e9)

```python
import json
from openpyxl import Workbook
import requests

response = requests.get(
    "http://10.0.0.14:7912/api/v1/spool?allow_archived=false&limit=9999&offset=0&sort=id:asc")
data = response.json()

wb = Workbook()
ws = wb.active
ws.title = "Spool"

# 定义列名
headers = [
    "id", "registered",
    "filament_id", "filament_registered", "filament_name",
    "filament_material", "filament_price", "filament_density",
    "filament_diameter", "filament_weight", "filament_color_hex",
    "vendor_id", "vendor_registered", "vendor_name",
    "remaining_weight", "initial_weight", "used_weight",
    "remaining_length", "used_length", "location", "archived"
]

ws.append(headers)


def safe_get(value):
    return "<null>" if value is None else value


# 写入数据
for item in data:
    filament = item.get("filament", {})
    vendor = filament.get("vendor", {})
    row = [
        safe_get(item.get("id")),
        safe_get(item.get("registered")),
        safe_get(filament.get("id")),
        safe_get(filament.get("registered")),
        safe_get(filament.get("name")),
        safe_get(filament.get("material")),
        safe_get(filament.get("price")),
        safe_get(filament.get("density")),
        safe_get(filament.get("diameter")),
        safe_get(filament.get("weight")),
        safe_get(filament.get("color_hex")),
        safe_get(vendor.get("id")),
        safe_get(vendor.get("registered")),
        safe_get(vendor.get("name")),
        safe_get(item.get("remaining_weight")),
        safe_get(item.get("initial_weight")),
        safe_get(item.get("used_weight")),
        safe_get(item.get("remaining_length")),
        safe_get(item.get("used_length")),
        safe_get(item.get("location")),
        safe_get(item.get("archived"))
    ]
    ws.append(row)

# 保存为 Excel 文件
wb.save("Spool.xlsx")
print("写入完成：Spool.xlsx")

```

**Comment by prevail90** [comment]
I am trying to use a label printer but for some reason when I print direct, it just prints blank. but when I download and print the images, it prints just fine.

**Reply by prevail90** [reply]
I think it is because it is a pdf instead of each being images.

**Reply by Ken6432** [reply]
Which label printer are you using?  I purchased a Niimbot B11 which allows me to print directly from the computer.  I defined my paper size in Spoolman to be the same as the label size, then print to the label printer in the same way as any other Windows program does

**Comment by ic9888** [comment]
Tried a few and struggled with too large margins, eventually got  agood result with Dymo 640CB using 19mm labels and modifying the windows page size options as well as the spoolman print options to get a label that fits nicely in the bambu spool label cut out.

**Comment by beikeland** [comment]
I considered getting a dedicated direct thermal Phomemo or Niimbot but picked up another surplus thermal transfer Zebra instead, I had some issues getting 40x25mm labels printing correctly, the software support is not the best for (older) Zebra on MacOS so I opted to put the ZPL code for the label into the template area of spoolman, edited in the {templating tags} and hey presto the label preview can be copied and pasted as raw zpl to the printer. Which also solved the layout problems I would later get when switching to the round Ø25mm NFC tags and printing on the tags instead of doing tag + label. Gonna look into something like Rino010 did so I don't have to do the copy/paste thing. The 40x25 template stuff is shown in https://github.com/Donkie/Spoolman/issues/606#issuecomment-3480474729 for inspiration.

**Comment by kusmi** [comment]
I'm considering a thermal printer as well - but how is your experience when you dry your filament? 

ABS/ASA has quite high drying temperatures, so I suppose thermal labels will just go "black" on those high temperatures?

**Comment by beikeland** [comment]
Yeah, I have some of the original manufacturer labels just to blank (they probably went all black, and continued exposure to temperature makes them blank/white)

My Spoolman labels are thermal transfer, not direct thermal. The kind that uses wax or resin ribbons that get transferred to the labels, and the resin is basically indestructible to all but the harshest chemicals so if you plan on doing higher temp drying I would definitively look into thermal transfer but the price is higher, even if you find some industrial surplus. Lots of the older 200dpi stuff available and thats plenty for our needs imho.

**Comment by Christian-B-ger** [comment]
i use a cheap niimbot B1, i export as a grafic and load that into the nimbot software. not bstraight forward but that works and the labels are relatively cheap

**Comment by Ken6432** [comment]
I have switched to a B1 now as well.  I am able to print directly to the printer from my Win11 computer.  In Spoolman, I defined a page size equal to the label.  When I hit Print in Spoolman, it goes to the Windows print dialog where I select the B1.  It has worked very well now for a few months.  No exporting graphics or PDFs needed.


> i use a cheap niimbot B1, i export as a grafic and load that into the nimbot software. not bstraight forward but that works and the labels are relatively cheap
>

**Reply by Christian-B-ger** [reply]
nice, how did you managed to get that as a printer. i read the stuff about the "driver" but i thought that will only be useless stuff
when i add that to windows it says "driver not available"...

**Reply by Ken6432** [reply]
I had to download the driver and it was very confusing and convoluted to get installed.  The instructions were awful, but I finally managed to get it installed.  I think I got to the download page from their support page

**Reply by guitarhead2000** [reply]
Is this for setting up 1 label at a time?  Have you selected multiple labels to print at once?  Looking to getting a B1 and havent found much online about how to set up Spoolman to print to it other than this thread.

**Reply by Ken6432** [reply]
I can select as many spools as I want.  I have mine set up to print 3 copies of each label, so I will quite often be printing 15 or so labels at a time.  Once the Niimbot printer driver gets installed (which as I said is very tedious) it behaves just like any other Windows printer.  I set the page size in Spoolman to be the same size as my labels in the B1.

I'd list how to install the driver, but honestly, I can't remember what the final process was.  I poked around with it for a few hours trying all kinds of install orders and uninstalls.  Finally, it just happened to work.

**Reply by guitarhead2000** [reply]
ok so you aren't saving each label as an image, just clicking print. gotcha. I think im gonna get one. I only have 50 spools but im sure i will find many more uses for it. thanks

**Comment by sarelon** [comment]
Datamax-O'Niel W1110 I picked up surplus printing 4x2 labels.  Only downside I really have is the printer wants 1" label spool cores and a max 5" label roll which are expensive.  I buy 3" cores and much bigger rolls and use a modified arbor for my spool rewinder to rewind the labels to 3d printed 1" cores.
